pkg/services/folder/folderimpl/folder.go

```diff
@@ -283,11 +283,18 @@ func (s *Service) Create(ctx context.Context, cmd *folder.CreateFolderCommand) (
 
 	var nestedFolder *folder.Folder
 	if s.features.IsEnabled(featuremgmt.FlagNestedFolders) {
+		// when uid is in the input, use the input uid for nested folder, otherwise use the uid of the legacy folder
+		var uid string
+		if cmd.UID != "" {
+			uid = cmd.UID
+		} else {
+			uid = dash.Uid
+		}
 		cmd := &folder.CreateFolderCommand{
 			// TODO: Today, if a UID isn't specified, the dashboard store
 			// generates a new UID. The new folder store will need to do this as
 			// well, but for now we take the UID from the newly created folder.
-			UID:         dash.Uid,
+			UID:         uid,
```

Comment:
I don't think we need this block; it's covered by [this](https://github.com/grafana/grafana/blob/d65899de7b1ff27fe50264a35dafa3b825f74eec/pkg/services/folder/folderimpl/folder.go#L220)

as I rule of thumb both entries in the two tables they need to have the same `UID`

Reply:
i have impression that the mock is not working properly, we might able to discuss in the 1-1, whatever i set as response value for savedashboard, the dash object is always empty...